### PR TITLE
implement autoscroll

### PR DESCRIPTION
**Pull request by hyusap**
solves #92

### CODE REVIEW

File: www/app/page.tsx [hunk 1]
```diff
@@ -146,6 +156,19 @@ export default function Home() {
...
+    const messageContainer = messageContainerRef.current;
+    if (!messageContainer) return;
+
+    messageContainer.addEventListener("scroll", () =>
+      setIsAtBottom(
+        Math.round(
+          messageContainer.scrollHeight - messageContainer.scrollTop
+        ) === messageContainer.clientHeight
+      )
+    );
```
Comment (VVoruganti) [inline www/app/page.tsx:169]: Make sure you add a cleanup function here such as 
```jsx
return () => {
    messageContainer.removeEventListener("scroll" , func)
}
```

File: www/app/page.tsx [hunk 2]
```diff
@@ -207,6 +230,11 @@ export default function Home() {
 
     const reader = data.body?.pipeThrough(new TextDecoderStream()).getReader()!;
 
+    const messageContainer = messageContainerRef.current;
```
Comment (VVoruganti) [inline www/app/page.tsx:233]: So from what I can tell about the logic here you are first scrolling too the bottom and then if the user scrolls up during rendering it won't continue to scroll. I'm wondering if that should be the behavior or just default to always checking

**Comment by VVoruganti** [comment]
Another behavior we should add here is when you switch conversations it should scroll too the bottom of the new conversation

File: www/app/page.tsx (hunk 2 above)
Comment (hyusap) [reply www/app/page.tsx:245]: i think this is standard:

- if you're at the bottom, you stay at the bottom,
- if you're reading something above, you stay above
- you can break out of the bottom lock at all times, by scrolling up

**Review by VVoruganti — APPROVED** [review]

**Pull request merged**